eng/scripts/Invoke-GeneratAndBuildV2.ps1

```diff
@@ -0,0 +1,43 @@
+#Requires -Version 7.0
+param (
+  [string]$inputJsonFile,
+  [string]$outputJsonFile
+)
+
+. (Join-Path $PSScriptRoot automation GenerateAndBuildLib.ps1)
+
+$inputJson = Get-Content $inputJsonFile | Out-String | ConvertFrom-Json
+$swaggerDir = $inputJson.specFolder
+$swaggerDir = $swaggerDir -replace "\\", "/"
+$readmeFiles = $inputJson.relatedReadmeMdFiles
+$commitid = $inputJson.headSha
+$repoHttpsUrl = $inputJson.repoHttpsUrl
+
+$generatedSDKPackages = New-Object 'Collections.Generic.List[System.Object]'
+
+foreach ( $relateReadmeFile in $readmeFiles ) {
+    $readmeFile = $relateReadmeFile -replace "\\", "/"
+    
+    $sdkPath =  (Join-Path $PSScriptRoot .. ..)
+    $sdkPath = Resolve-Path $sdkPath
+    $sdkPath = $sdkPath -replace "\\", "/"
+
+    
+    $readme = ""
+    if ($commitid -ne "") {
```

Comment:
Tip: not necessary to compare. PowerShell is very truthy, so just `if ($commitid) {` checks that `$commitid` is not null or an empty string / collection. Same can be said for other comparisons below.